### PR TITLE
Sam tov rdf adf fix

### PR DESCRIPTION
**Pull request by SamTov**
Fix # 248 

#### Summary of additions and changes

* Remove any non-mini-batched rdf code
* split code into modules
* Fix excess variable assignment
* Fix type errors
* Add species-wise calculation capability to RDF
* Clean the adf code
* Add functionality (not currently working but the infrastructure is there) to multiply adf by a pre-factor.
* Allow for species-wise calculations in the ADF via a species argument e.g. species=["N", "C"].

I have tested all of this code on a system of EMIM-F using both tf and non tf and using both species specification and no species specification.

#### How to test this pull request
Run any RDF , ADF computation.

### CODE REVIEW

File: mdsuite/calculators/radial_distribution_function.py [hunk 1]
```diff
@@ -59,7 +62,7 @@ class RadialDistributionFunction(Calculator, ABC):
     loaded_property : str
             Property loaded from the database_path for the analysis
     minibatch: int, default None
-            Size of a individual minibatch, if set. By default minibatching is not applied
+            Size of a individual minibatch, if set. By default mini-batching is not applied
 
     See Also
     --------
```
Comment (PythonFZ) [inline mdsuite/calculators/radial_distribution_function.py:68]: the example below should be changed to RDF from ADF

File: mdsuite/calculators/radial_distribution_function.py [hunk 2]
```diff
@@ -322,12 +352,12 @@ def _load_positions(self, indices: list) -> tf.Tensor:
...
                 tf.Tensor of tensor_values loaded from the hdf5 database_path
         """
-
-        if self.molecules:
-            path_list = [join_path(species, "Positions") for species in self.experiment.molecules]
-            return self.experiment.load_matrix("Positions", path=path_list, select_slice=np.s_[:, indices])
+        path_list = [join_path(species, "Positions") for species in self.species]
+        data = self.experiment.load_matrix("Positions", path=path_list, select_slice=np.s_[:, indices])
+        if len(self.species) == 1:
+            return tf.concat([data], axis=0)
```
Comment (PythonFZ) [inline mdsuite/calculators/radial_distribution_function.py:358]: `tf.concat([data], axis=0)` == `data`
you should be able to just return data

**Review by PythonFZ — COMMENTED** [review]
I briefly looked over it. I'll do a full review when all changes are made.

File: mdsuite/calculators/radial_distribution_function.py [hunk 3]
```diff
@@ -265,10 +295,10 @@ def _correction_2(data: np.array) -> np.array:
                     result of the operation
 
             """
-
```
[the remark this answers is not in the document]
Comment (SamTov) [reply mdsuite/calculators/radial_distribution_function.py:268]: Currently the corrections don't work and I have no clue why. However, there are actually no publications on performing these over half box calculations.

File: mdsuite/calculators/radial_distribution_function.py (hunk 1 above)
Comment (SamTov) [reply mdsuite/calculators/radial_distribution_function.py:68]: Yep.

File: mdsuite/calculators/radial_distribution_function.py (hunk 2 above)
Comment (SamTov) [reply mdsuite/calculators/radial_distribution_function.py:358]: Yes! I have changed it.

File: mdsuite/calculators/radial_distribution_function.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (SamTov) [reply mdsuite/calculators/radial_distribution_function.py:268]: If you try it and plot the result, you will find that the whole g(r) is shifted by 1. The structure is actually correct so the evaluation works it is just off by 1. I have no clue why and haven't found any good solution.

File: mdsuite/calculators/angular_distribution_function.py [hunk 4]
```diff
@@ -158,105 +218,186 @@ def _load_positions(self, indices: list) -> tf.Tensor:
...
+    def _prepare_triples_generator(self):
+        """
+        Prepare the triples generators including tf.function
+        Returns
+        -------
+
+        """
         if self.use_tf_function:
-            @tf.function
+            @tf.function()
```
Comment (PythonFZ) [inline mdsuite/calculators/angular_distribution_function.py:270]: I think that the brackets aren't required

File: mdsuite/calculators/angular_distribution_function.py [hunk 5]
```diff
@@ -158,105 +218,186 @@ def _load_positions(self, indices: list) -> tf.Tensor:
...
-                    yield tf.concat(self._load_positions(idx), axis=0)  # Load th
-                # yield self._load_positions(idx)  # <- bottleneck, because loading single configurations
+        return _get_triplets
 
-        dataset = tf.data.Dataset.from_generator(generator, output_signature=(
-            tf.TensorSpec(shape=(None, None), dtype=tf.float32)
-        ))
+    def _compute_rijk_matrices(self, tmp: tf.Tensor, timesteps: int):
+        """
+        Compute the rij matrix.
```
Comment (PythonFZ) [inline mdsuite/calculators/angular_distribution_function.py:289]: function is called rijk but docstring states rij

File: mdsuite/utils/linalg.py [hunk 6]
```diff
@@ -56,4 +56,4 @@ def get_angles(r_ij_mat, indices, acos=True):
     r_ij = tf.gather_nd(r_ij_mat, tf.stack([indices[:, 0], indices[:, 1], indices[:, 2]], axis=1))
     r_ik = tf.gather_nd(r_ij_mat, tf.stack([indices[:, 0], indices[:, 1], indices[:, 3]], axis=1))
 
-    return angle_between(r_ij, r_ik, acos)
+    return angle_between(r_ij, r_ik, acos), 1 / (abs(r_ij) * abs(r_ik))
```
Comment (PythonFZ) [inline mdsuite/utils/linalg.py:59]: I asume this is for the potential implementation of the prefactor?

File: mdsuite/calculators/angular_distribution_function.py (hunk 4 above)
Comment (SamTov) [reply mdsuite/calculators/angular_distribution_function.py:270]: Oh yes sorry, I was experimenting with the compile and the graph_mode stuff.

File: mdsuite/utils/linalg.py (hunk 6 above)
Comment (SamTov) [reply mdsuite/utils/linalg.py:59]: Yes but in the actual code I set the pre-factor to 1

**Pull request merged**